tests/algorithms/test_bounded_standard_deviation.py

```diff
@@ -8,5 +8,5 @@ def test_c_api(self):
         lower_bound, upper_bound = 0, 15
         bsd = dp.BoundedStandardDeviation(epsilon, lower_bound, upper_bound)
         result = bsd.result(example_data)
-        assert type(result) is float
-        assert lower_bound <= result <= upper_bound
+        assert type(result) is float and result >= 0
+        assert result <= (upper_bound - lower_bound) / 2
```

Comment:
Is this again based on Popoviciu's inequality

Reply:
I don't know what's the usual proof, but here's how you can derive it from Popoviciu's inequality on variances (I probably made it too complicated):

&sigma;<sup>2</sup> &le; 1/4 (M - m)<sup>2</sup>

0 &le; (1/2 (M - m))<sup>2</sup> - &sigma;<sup>2</sup>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# a<sup>2</sup> - b<sup>2</sup> = (a + b)(a - b)

0 &le; (1/2 (M - m) + &sigma;)  (1/2 (M - m) - &sigma;)&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# divide by (1/2 (M - m) + &sigma;) which is &ge; 0 because M &ge; m and &sigma; &ge; 0

0 &le; 1/2 (M - m) - &sigma;

&sigma; &le; 1/2 (M - m)

Reply:
I suspected that; just wanted to verify